### PR TITLE
Refactor zendesk implementation with tests

### PR DESCRIPTION
**Pull request by benthorner**
https://trello.com/c/i5iX0OiN/224-zendesk-integration-is-broken

The previous implementation used a set of client classes (models?!),
assigned to a global constant, which introduced an unnecessary amount
of indirection. Instead, we now have a Zendesk module that can return a
real or dummy client as appropriate.

Note that Zendesk will now be enabled for all non-dev/test environments,
as we should allow for manual testing in these environments. This will require populating the appropriate environment variables with the production secrets.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -18,21 +19,6 @@
     delete_index
   end
 
-  config.expect_with :rspec do |expectations|
-    expectations.include_chain_clauses_in_custom_matcher_descriptions = true
-  end
-
-  config.mock_with :rspec do |mocks|
```
Comment (andrewgarner) [inline spec/spec_helper.rb]: How is this related?

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -18,21 +19,6 @@
...
-    mocks.verify_partial_doubles = true
-  end
-
-  config.shared_context_metadata_behavior = :apply_to_host_groups
```
Comment (andrewgarner) [inline spec/spec_helper.rb]: Again, how is this related?

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -18,21 +19,6 @@
...
-
-  if config.files_to_run.one?
```
Comment (andrewgarner) [inline spec/spec_helper.rb]: Again, how is this related?

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -18,21 +19,6 @@
...
-    config.default_formatter = "doc"
-  end
-
-  config.profile_examples = 1
```
Comment (andrewgarner) [inline spec/spec_helper.rb]: Again, how is this related?

Comment (benthorner) [reply spec/spec_helper.rb]: Indeed, it's not. I'm trying to feel where the 'cleaning up a file can go in the same PR' line is.
